docs/catalog/datasource.md

```diff
@@ -0,0 +1,317 @@
+
+### `description` (String)
+
+The description property hold human-readable text to describe the datasource. Druid does not
+yet use this property, though the intent is that the Druid Console will display this text
+in some useful way.
+
+```json
+{
+  "type" : "datasource",
+  "properties" : {
+     "description" : "Web server performance metrics"
+  }
+}
+```
+
+### `segmentGranularity` (String)
+
+Defines the primary partitioning for the datasources.
+Tells MSQ the size of the time chunk that each segment is to cover. This is the equivalent of
+the MSQ `PARTITIONED BY` clause. Provide the granularity
+as an ISO 8601 time period format that matches one of Druid's supported segment granularities.
+For example, `PT1H` or `P1D`. Once you specify this property, you no longer need the
+`PARTITIONED BY` clause.
+
+Example:
+
+```json
+{
+  "type" : "datasource",
+  "properties" : {
+     "segmentGranularity" : "PT1H"
+  }
+}
+```
+
+### `clusterKeys` (List of Objects)
+
+This property identifies the secondary partitioning for a datasource. This property is optional
+if you do not need secondary partitioning. Else, it is a list of pairs of column names and
+sort directions. Ascending is the default if you do not specify a direction.
+
+Example:
+
+```json
+{
+  "type" : "datasource",
+  "properties" : {
+    "clusterKeys" : [ {
+      "column" : "a"
+    }, {
+      "column" : "b",
+      "desc" : true
+    } ]
+  }
+}
+```
+
+The two fields per cluster key are:
+
+* `column`: the name of a column. (The name need not be defined in table metadata, but it
+  must exist as part of the MSQ query at ingest time.)
+* `desc`: a Boolean value. `true` means descending, `false` (or omitted) means ascending.
+
+### `targetSegmentRows` (Integer)
+
+Druid normally creates segments with 5 million rows. However, if your rows are especially small, you
+may find it useful to create segments with more rows. If your rows are unusually large, then you
+may want to create segments with fewer rows. Use this property to control the target row count.
+(Actual segments will be limited by actual data, and will be of roughly the requested size.)
+Omit this property unless you have a special need: Druid will use its default.
+
+Example:
+
+
+```json
+{
+  "type" : "datasource",
+  "properties" : {
+     "segmentGranularity" : "PT1H"
+  }
+}
+```
+
+### `hiddenColumns` (List of Strings)
+
+Druid does not provide an easy way to drop columns from existing datasources. Dropping a column
+can be done with compaction specs, but is awkward and costly for large data sets. If your data
+has a limited lifetime (that is, it expires after a month or two), then there is a simpler, more
+efficient solution: simply hide the existing columns. Hidden columns will not appear in SQL
+queries, though the data still exists and is visible to native queries. You hide a column by
+adding it to the `hiddenColumns` list.
+
+Example:
+
+```json
+{
+  "properties": {
+    "targetSegmentRows" : 1000000
```

Comment:
this example is not correct, copy paste error?

Reply:
Looks OK... This is a partial example: it doesn't have a type. I removed the outer brackets to be clearer that this is a partial example.